dockerversion/useragent.go

```diff
@@ -5,11 +5,23 @@ import (
 
 	"github.com/docker/docker/pkg/parsers/kernel"
 	"github.com/docker/docker/pkg/useragent"
+	"golang.org/x/net/context"
 )
 
+// uaStringKey is used as key type for user agent string in net/context struct
+type uaStringKey int
```

Comment:
After talking to @icecrime, I am going to put the custom type in `api/server` and use it in `distribution/registry.go` (which already depends on `api/server`).  Thus, when `api/server` gets moved out into `docker/engine-api` at some point in the future, there will not be any dependency on `dockerversion`.

@calavera If this plan doesn't satisfy your concerns, can you elaborate?